jax/_src/basearray.py

```diff
@@ -187,4 +186,3 @@ def copy_to_host_async(self):
   np.ndarray,  # NumPy array type
   StaticScalar,  # valid scalars
 ]
```

Comment:
Could we still define this for older versions? It would be a shame to lose this information on the doc page: https://docs.jax.dev/en/latest/_autosummary/jax.typing.ArrayLike.html

Reply:
yeah, that will be bad. How about this:
```python
if sys.version_info.major == 3 and sys.version_info.minor < 14:
  # Python 3.14 raises
  # AttributeError: 'typing.Union' object attribute '__doc__' is read-only
  StaticScalar.__doc__ = "Type annotation for JAX-compatible static scalars."
```

Reply:
I think I recall that if you put a string literal below the object definition, sphinx will treat it as the docstring. It would be worth checking if that works here

Reply:
I was pointed out to this example:
- https://narwhals-dev.github.io/narwhals/api-reference/typing/#narwhals.typing.IntoExpr
- https://github.com/narwhals-dev/narwhals/blob/9da549e86f83c96df19ab71916e7d4335c89c1e3/docs/api-reference/typing.md?plain=1#L6-L15
- https://github.com/narwhals-dev/narwhals/blob/9da549e86f83c96df19ab71916e7d4335c89c1e3/narwhals/typing.py#L51-L58

I'm not yet able to make this work with Sphinx

Reply:
Maybe, for now we can skip `*.__doc__` mutation with python version guard. Docs won't be affected for some time as docs will be built with python<3.14.

Reply:
Sounds good.